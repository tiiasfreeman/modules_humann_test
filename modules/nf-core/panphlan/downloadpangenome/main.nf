process PANPHLAN_DOWNLOADPANGENOME {
    label 'process_high'

    conda "${moduleDir}/environment.yml"
    container "${ workflow.containerEngine == 'singularity' && !task.ext.singularity_pull_docker_container ?
        'https://depot.galaxyproject.org/singularity/mulled-v2-e3818ba00c0f15bcda92c714b4fa500c626067fe:cae224e6965f9abd8b08407231aff3bba632f1de-0':
        'biocontainers/mulled-v2-e3818ba00c0f15bcda92c714b4fa500c626067fe:cae224e6965f9abd8b08407231aff3bba632f1de-0' }"

    input:
    val(species_name)

    output:
    path("indexes")                       , emit: indexes
    path("${prefix}/*_pangenome.tsv")     , emit: pangenome
    path "versions.yml"                   , emit: versions

    when:
    task.ext.when == null || task.ext.when

    script:
    def args = task.ext.args ?: ''
    prefix = "${species_name}"
    """
    panphlan_download_pangenome.py \\
        -i ${species_name} \\
        -o . \\
        ${args}

    mkdir indexes
    mv ${prefix}/*.bt2 indexes

    cat <<-END_VERSIONS > versions.yml
    "${task.process}":
        panphlan: \$(samtools --version |& sed '1!d ; s/samtools //')
    END_VERSIONS
    """

    stub:
    def args = task.ext.args ?: ''
    """
    touch ${prefix}.bam

    cat <<-END_VERSIONS > versions.yml
    "${task.process}":
        panphlan: \$(samtools --version |& sed '1!d ; s/samtools //')
    END_VERSIONS
    """
}
