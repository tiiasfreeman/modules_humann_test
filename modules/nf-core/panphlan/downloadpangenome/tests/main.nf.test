nextflow_process {

    name "Test Process PANPHLAN_DOWNLOADPANGENOME"
    script "modules/nf-core/panphlan/downloadpangenome/main.nf"
    process "PANPHLAN_DOWNLOADPANGENOME"

    test("Test download B. fragilis pangenome") {

        when {

            process {
                """
                input[0] = "Bacteroides_fragilis"
                """
            }
        }

        then {
            assertAll (
            { assert process.success },
            { assert snapshot(process.out).match() }
            )
        }

    }

    test("Test download B. fragilis pangenome '-stub' ") {

            options "-stub"

        when {

            process {
                """
                input[0] = "Bacteroides_fragilis"
                """
            }
        }

        then {
            assertAll (
            { assert process.success },
            { assert snapshot(process.out).match() }
            )
        }
    }
    
}