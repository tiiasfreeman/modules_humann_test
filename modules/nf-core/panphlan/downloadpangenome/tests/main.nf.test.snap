{
    "Test download B. fragilis pangenome '-stub' ": {
        "content": [
            {
                "0": [
                    [
                        "Bacteroides_fragilis.bt2:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "1": [
                    "Bacteroides_fragilis_pangenome.tsv:md5,d41d8cd98f00b204e9800998ecf8427e"
                ],
                "2": [
                    "versions.yml:md5,0f1949425375d8c603081ed92e57d83f"
                ],
                "indexes": [
                    [
                        "Bacteroides_fragilis.bt2:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "pangenome": [
                    "Bacteroides_fragilis_pangenome.tsv:md5,d41d8cd98f00b204e9800998ecf8427e"
                ],
                "versions": [
                    "versions.yml:md5,0f1949425375d8c603081ed92e57d83f"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "24.04.2"
        },
        "timestamp": "2024-06-10T17:50:00.930495226"
    },
    "Test download B. fragilis pangenome": {
        "content": [
            {
                "0": [
                    [
                        "Bacteroides_fragilis.1.bt2:md5,3817e33332a6c3c8c21589bac2c0f2df",
                        "Bacteroides_fragilis.2.bt2:md5,573db3ab72de9ff8e52e7b8452518488",
                        "Bacteroides_fragilis.3.bt2:md5,53f57747525c87b19cec519b550f7c9a",
                        "Bacteroides_fragilis.4.bt2:md5,c5dc03141827afa22e0a6c18e3522b97",
                        "Bacteroides_fragilis.rev.1.bt2:md5,6488a3b459c54219fa95f8d144fc95c0",
                        "Bacteroides_fragilis.rev.2.bt2:md5,57869f321c3534d68ee2b6e437b630b7"
                    ]
                ],
                "1": [
                    "Bacteroides_fragilis_pangenome.tsv:md5,1efc51187d49ef33f23894f64c6c2d47"
                ],
                "2": [
                    "versions.yml:md5,0f1949425375d8c603081ed92e57d83f"
                ],
                "indexes": [
                    [
                        "Bacteroides_fragilis.1.bt2:md5,3817e33332a6c3c8c21589bac2c0f2df",
                        "Bacteroides_fragilis.2.bt2:md5,573db3ab72de9ff8e52e7b8452518488",
                        "Bacteroides_fragilis.3.bt2:md5,53f57747525c87b19cec519b550f7c9a",
                        "Bacteroides_fragilis.4.bt2:md5,c5dc03141827afa22e0a6c18e3522b97",
                        "Bacteroides_fragilis.rev.1.bt2:md5,6488a3b459c54219fa95f8d144fc95c0",
                        "Bacteroides_fragilis.rev.2.bt2:md5,57869f321c3534d68ee2b6e437b630b7"
                    ]
                ],
                "pangenome": [
                    "Bacteroides_fragilis_pangenome.tsv:md5,1efc51187d49ef33f23894f64c6c2d47"
                ],
                "versions": [
                    "versions.yml:md5,0f1949425375d8c603081ed92e57d83f"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "24.04.2"
        },
        "timestamp": "2024-06-07T19:05:31.33300438"
    }
}